src/Reflection/RequireExtension/RequireExtendsMethodsClassReflectionExtension.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\RequireExtension;
+
+use PHPStan\Analyser\OutOfClassScope;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\MethodsClassReflectionExtension;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\VerbosityLevel;
+
+class RequireExtendsMethodsClassReflectionExtension implements MethodsClassReflectionExtension
+{
+
+	/** @var array<string, array<string, true>> */
+	private array $inProcess = [];
+
+	public function hasMethod(ClassReflection $classReflection, string $methodName): bool
+	{
+		return $this->findMethod($classReflection, $methodName) !== null;
+	}
+
+	public function getMethod(ClassReflection $classReflection, string $methodName): MethodReflection
+	{
+		$method = $this->findMethod($classReflection, $methodName);
+		if ($method === null) {
+			throw new ShouldNotHappenException();
+		}
+
+		return $method;
+	}
+
+	private function findMethod(ClassReflection $classReflection, string $methodName): ?MethodReflection
+	{
+		if (!$classReflection->isInterface()) {
+			return null;
+		}
+
+		$extendsTags = $classReflection->getRequireExtendsTags();
+		foreach ($extendsTags as $extendsTag) {
+			$type = $extendsTag->getType();
+
+			$typeDescription = $type->describe(VerbosityLevel::typeOnly());
+			if (isset($this->inProcess[$typeDescription][$methodName])) {
+				continue;
+			}
+
+			$this->inProcess[$typeDescription][$methodName] = true;
+
+			if (!$type->hasMethod($methodName)->yes()) {
+				unset($this->inProcess[$typeDescription][$methodName]);
```

Comment:
Is this recursion guard tested?

Reply:
I tried thinking of a way this could get recursive, but wasn't able to come up with a example.

do you think its impossible and the recursion-guard should be removed?

Reply:
I'd expect something like this to end up in infinite recursion https://phpstan.org/r/aa768c71-4596-4edf-a900-60e4db58582b

Reply:
thx for the inspiration. I tried similar things before and I can't force them to endless recurse.

I think this might not happen, because the new extension early exits for non-interfaces and the `phpstan-require-extends` requires a class as arguments.

since I was still not able to force a recursion, I have removed the recursion guards for now